bundles/http/Citrix/Canvas-LMS/0b9f4444-56d6-4b4e-b108-5624645e71d2.js

```diff
@@ -0,0 +1,352 @@
+    let announcementCondition, announcementLink, announcementIncrement = 1;
+    //announcement pagination
+    do {
+        announcementCondition = "";
+        let announcementResponse = await serviceClient.fetch(`api/v1/announcements?context_codes[]=course_${courseId}&start_date=${StartDate}&end_date=${EndDate}&page=${announcementIncrement}&per_page=100`);  //Announcement API call
+        announcementIncrement++
+        let tempAnnouncementLink = announcementResponse.headers.map.link;
+        tempAnnouncementLink = tempAnnouncementLink.split(",");
+
+        let enrollmentCondition, enrollmentLink, enrollmentIncrement = 1;
+        //enrollment pagination
+        do {
+            enrollmentCondition = "";
+            let enrollmentResponse = await serviceClient.fetch(`/api/v1/courses/${courseId}/enrollments?page=${enrollmentIncrement}&per_page=100`);  //Enrollments API call
+            enrollmentIncrement++
+            let tempEnrollmentLink = enrollmentResponse.headers.map.link;
+            tempEnrollmentLink = tempEnrollmentLink.split(",");
+            var enrollment;
+            if (enrollmentResponse.ok) {
+                enrollment = await enrollmentResponse.json();
+                const promise = await enrollment.map(async enrollmentValue => {                
+                    let enrollmentBody = {
+                        "id": enrollmentValue.id,
+                        "course_id": enrollmentValue.course_id,
+                        "role": enrollmentValue.role,
+                        "user_login_id": enrollmentValue.user.login_id,
+                        "enrollment_state": enrollmentValue.enrollment_state,
+                        "user_id": enrollmentValue.user.id,
+                        "user_name": enrollmentValue.user.name
+                    }
+                    await dataStore.save('enrollments', JSON.stringify(enrollmentBody))
+                })
+                await Promise.all(promise)
+            } else {
+                throw new Error(`Enrollments sync failed ${enrollmentResponse.status}:${enrollmentResponse.statusText}.`)
+            }
+
+            if (announcementResponse.ok) {
+                let announcement = await announcementResponse.json();
+                const promise = await announcement.map(async announcementValue => {
+                    const promise = await enrollment.map(async enrollmentValue => {
+                        let announcementBody = {
+                            "id": announcementValue.id,
+                            "enrollme_id": enrollmentValue.id,
+                            "enrollme_course_id": enrollmentValue.course_id,
+                            "title": announcementValue.title,
+                            "message": announcementValue.message,
+                            "enrollme_user_login_id": enrollmentValue.user.login_id,
+                            "enrollme_role": enrollmentValue.role,

```

Comment:
Imho you can use a relationship here. You don't need to put all values here if you have them in `enrollments` table.
See https://developer.cloud.com/citrixworkspace/citrix-workspace-platform/build-workspace-microapp-integrations/docs/script-development#relationships

Reply:
The relationship doesn't work in notification because of https://support.citrix.com/article/CTX286264. So we are pushing these details in 2 different tables.

Reply:
I didn't know about it. Ok, then.